tests/conftest.py

```diff
@@ -385,3 +386,26 @@ def in_memory_pip():
 def deprecated_python():
     """Used to indicate whether pip deprecated this python version"""
     return sys.version_info[:2] in [(2, 7)]
+
+
+@pytest.fixture(scope="session")
+def cert_factory(tmpdir_factory):
+    def factory():
+        # type: () -> str
+        """Returns path to cert/key file.
+        """
+        output_path = Path(str(tmpdir_factory.mktemp("certs"))) / "cert.pem"
+        # Must be Text on PY2.
+        cert, key = make_tls_cert(u"localhost")
+        with open(str(output_path), "wb") as f:
+            f.write(serialize_cert(cert))
+            f.write(serialize_key(key))
+
+        return str(output_path)
+
+    return factory
+
+
+@pytest.fixture
+def ssl_context(cert_factory):
```

Comment:
This fixture doesn't seem to be used (and doesn't do much ^^)

Reply:
Ah, that's true. :) Fixed in [`204a004`](https://github.com/pypa/pip/pull/7289/commits/204a0043776cc3e0cc74841c051c8da831c2ca4b#diff-6950ca48019ac71dcd4978913438299bR53).